### PR TITLE
implement `gt kill`

### LINKED ISSUE #824: implement `gt kill`

### PR DESCRIPTION
**Pull request by charlierudolph**
resolves #824 

I changed the order of some of the commands to prevent the need for a two pushes in the undo process.

### CODE REVIEW

File: lib/git/branch.go [hunk 1]
```diff
@@ -73,6 +79,18 @@ func HasBranch(branchName string) bool {
 	return false
 }
 
+func HasLocalBranch(branchName string) bool {
+	output := util.GetCommandOutput("git", "branch")
```
Comment (kevgo) [inline lib/git/branch.go]: Extract a `GetLocalBranches` function?

File: cmd/kill.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+	if len(args) == 0 {
+		result.TargetBranch = result.InitialBranch
+	} else {
+		result.TargetBranch = args[0]
+	}
+
+	config.EnsureIsFeatureBranch(result.TargetBranch, "You can only kill feature branches.")
+
+	result.IsTargetBranchLocal = git.HasLocalBranch(result.TargetBranch)
```
Comment (charlierudolph) [inline cmd/kill.go]: The kill command always has a target branch though and the point of the variable is whether or not it is local.

File: lib/config/config.go [hunk 3]
```diff
@@ -19,14 +20,24 @@ func CompileAncestorBranches(branchName string) (result []string) {
 	}
 }
 
-func DeleteAncestorBranches(branchName string) {
-	removeConfigurationValue("git-town-branch." + branchName + ".ancestors")
+func DeleteAllAncestorBranches() {
+	configs := util.GetCommandOutput("git", "config", "--get-regexp", "^git-town-branch\\..*\\.ancestors$")
```
Comment (charlierudolph) [inline lib/config/config.go]: I don't understand how that would help here. This is used for getting the configuration keys for all ancestor branches so they can be deleted

File: lib/git/branch.go (hunk 1 above)
Comment (charlierudolph) [inline lib/git/branch.go]: done

File: cmd/kill.go (hunk 2 above)
Comment (kevgo) [inline cmd/kill.go]: I see. Your name is better then.

File: lib/config/config.go (hunk 3 above)
Comment (kevgo) [inline lib/config/config.go]: I see. Good as is then.

**Pull request merged**